lib/cdo/contact_rollups.rb

```diff
@@ -138,31 +138,32 @@ def self.mysql_multi_connection
   ROLE_FORM_SUBMITTER = "Form Submitter".freeze
   CENSUS_FORM_NAME = "Census".freeze
 
-  def self.build_contact_rollups
+  def self.build_contact_rollups(log_collector)
     start = Time.now
 
-    initialize_connections_to_database_clone
+    log_collector.time!('initialize_connections_to_database_clone') {initialize_connections_to_database_clone}
 
     @@pegasus_clone_db_writer.run "SET SQL_SAFE_UPDATES = 0"
     # set READ UNCOMMITTED transaction isolation level on both read connections to avoid taking locks
     # on tables we are reading from during what can be multi-minute operations
     @@dashboard_clone_db_reader.run "SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED"
     ActiveRecord::Base.connection.execute "SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED"
-    create_destination_table
-    insert_from_pegasus_forms
-    insert_from_dashboard_contacts
-    insert_from_dashboard_pd_enrollments
-    insert_from_dashboard_census_submissions
-    update_geo_from_school_data
-    update_unsubscribe_info
-    update_roles
-    update_grades_taught
-    update_ages_taught
-    update_district
-    update_school
-    update_courses_facilitated
-    update_professional_learning_enrollment
-    update_professional_learning_attendance
+
+    log_collector.time!('create_destination_table') {create_destination_table}
+    log_collector.time!('insert_from_pegasus_forms') {insert_from_pegasus_forms}
+    log_collector.time!('insert_from_dashboard_contacts') {insert_from_dashboard_contacts}
+    log_collector.time!('insert_from_dashboard_pd_enrollments') {insert_from_dashboard_pd_enrollments}
+    log_collector.time!('insert_from_dashboard_census_submissions') {insert_from_dashboard_census_submissions}
+    log_collector.time!('update_geo_from_school_data') {update_geo_from_school_data}
+    log_collector.time!('update_unsubscribe_info') {update_unsubscribe_info}
+    log_collector.time!('update_roles') {update_roles}
+    log_collector.time!('update_grades_taught') {update_grades_taught}
+    log_collector.time!('update_ages_taught') {update_ages_taught}
+    log_collector.time!('update_district') {update_district}
+    log_collector.time!('update_school') {update_school}
+    log_collector.time!('update_courses_facilitated') {update_courses_facilitated}
+    log_collector.time!('update_professional_learning_enrollment') {update_professional_learning_enrollment}
+    log_collector.time!('update_professional_learning_attendance') {update_professional_learning_attendance}
```

Comment:
I feel like there _must_ be a cleaner way to write this, but I don't know what it is off the top of my head.  Maybe something like:

```ruby
%w(
    create_destination_table
    insert_from_pegasus_forms
    insert_from_dashboard_contacts
    insert_from_dashboard_pd_enrollments
    insert_from_dashboard_census_submissions
    update_geo_from_school_data
    update_unsubscribe_info
    update_roles
    update_grades_taught
    update_ages_taught
    update_district
    update_school
    update_courses_facilitated
    update_professional_learning_enrollment
    update_professional_learning_attendance
).each do |method_name|
  log_collector.time!(method_name) { send method_name }
end
```

Reply:
interesting use of `send` method. I'll take a look.

Reply:
I'm holding off making change here since we just need data to drive the next step of re-architecting this work. Will keep this suggestion in mind.